### ISSUE TITLE
pydrake cannot be consumed as a Bazel external; binaries do not work when called via `bazel (run|test) @drake//${target}`

### ISSUE THREAD

**Issue by EricCousineau-TRI** [RobotLocomotion/drake#7972]
For using Drake downstream, I was testing out running `@drake//tools:drake_visualizer` from the project, tried importing `pydrake`, and found that it could not be imported.

When I went to Drake, I tried running it such that the binary would be located in `bazel-bin/external/drake/...`, but got an error. To illustrate:
```
$ bazel run //bindings/pydrake:common_test
...
INFO: Running command line: bazel-bin/bindings/pydrake/common_test
...
OK

$ bazel run @drake//bindings/pydrake:common_test
INFO: Running command line: bazel-bin/external/drake/bindings/pydrake/common_test
...
    from ._common_py import *
ImportError: libdrake.so: cannot open shared object file: No such file or directory
```

Upon inspection, it does not appear to be linking the libraries correctly:
```
$ ldd bazel-bin/external/drake/bindings/pydrake/common_test.runfiles/drake/bindings/pydrake/_common_py.so | grep 'not found'
        libdrake.so => not found
        libgurobi75.so => not found
        libiomp5.so => not found
        libmosek64.so.7.1 => not found
        libvtkFiltersSources-8.0.so.1 => not found
...
        libignition_rndf.so => not found
        liblcm.so => not found
        libscsdir.so => not found
        libignition_math.so => not found

$ ldd bazel-bin/external/drake/bindings/pydrake/common_test.runfiles/drake/tools/install/libdrake/libdrake.so | grep 'not found'
        libignition_rndf.so => not found
        libignition_math.so => not found
        libvtkFiltersSources-8.0.so.1 => not found
...
        libgurobi75.so => not found
        libiomp5.so => not found
        libmosek64.so.7.1 => not found
        libscsdir.so => not found
        liblcm.so => not found
```

Not sure if this is an issue with either the `drake_pybind_*` macros, the `drake_cc_*` macros, or something with Bazel and externals in general.

As an aside, Bazel also rebuilds everything in Drake from scratch when going from `bazel run //${target}` to `bazel run @drake//${target}`.

**Comment by EricCousineau-TRI** [comment]
Filed upstream issue: https://github.com/bazelbuild/bazel/issues/4594

**Comment by EricCousineau-TRI** [comment]
Closed via #8240
